File: iOS/PanoAudioChat/PanoAudioChatSDK/Source/Sections/Setting/PanoVoiceTableController.m
//
//  PanoVoiceTableController.m
//  PanoAudioChat
//
//  Created by pano on 2021/7/22.
//

#import "PanoVoiceTableController.h"
#import "PanoClientService.h"

@interface PanoVoiceTableController ()

@end

@implementation PanoVoiceTableController {
    PanoAudioVoiceChangerOption selectedIndex;
}

- (void)viewDidLoad {
    [super viewDidLoad];
    selectedIndex = PanoClientService.service.audioVoiceChanger;
}

- (void)tableView:(UITableView *)tableView willDisplayCell:(UITableViewCell *)cell forRowAtIndexPath:(NSIndexPath *)indexPath {
    cell.accessoryType = indexPath.row == selectedIndex ? UITableViewCellAccessoryCheckmark : UITableViewCellAccessoryNone;
}


- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath {
    [tableView deselectRowAtIndexPath:indexPath animated:true];
    selectedIndex = indexPath.row;
    [tableView reloadData];
}
- (IBAction)finish:(id)sender{
    PanoClientService.service.audioVoiceChanger = selectedIndex;
    [self pop:nil];
}

- (IBAction)pop:(id)sender {
    [self dismissViewControllerAnimated:true completion:nil];
}
@end

File: iOS/PanoAudioChat/PanoAudioChat/Classes/Sections/Setting/PanoVoiceTableController.swift
//
//  PanoVoiceTableController.swift
//  PanoAudioChat
//
//  Created by wenhua.zhang on 2021/3/24.
//

import UIKit

class PanoVoiceTableController: UITableViewController {
    
    var selectedIndex = PanoClientService.audioVoiceChanger.rawValue;
    
    override func viewDidLoad() {
        super.viewDidLoad()
    }

    override func tableView(_ tableView: UITableView, willDisplay cell: UITableViewCell, forRowAt indexPath: IndexPath) {
        cell.accessoryType = indexPath.row == selectedIndex ? .checkmark : .none;
    }
    
    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        tableView.deselectRow(at: indexPath, animated: true);
        selectedIndex = indexPath.row
        tableView.reloadData()
    }
    
    @IBAction func finish(_ sender: Any) {
        PanoClientService.setAudioVoiceChangerMode(mode: PanoAudioVoiceChangerOption(rawValue: selectedIndex)!)
        self.dismiss(animated: true, completion: nil)
    }
    
    @IBAction func pop(_ sender: Any) {
        self.dismiss(animated: true, completion: nil)
    }
}
